### PR TITLE
Add docs/architecture.md

### PR DESCRIPTION
**Pull request by FabijanC**

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    - when a new contract is deployed, its metadata's IPFS hash is extracted from the end of the bytecode
+    - the metadata is fetched from IPFS
+    - source files information and compiler version are read from the metadata file
+    - the source files are fetched and recompiled
+    - files are stored to the repo
+- UI:
+  - provides a user friendly interface for communicating with the server
+  - two purposes:
+    - check if an address is verified
+    - verify a contract
```
Comment (chriseth) [inline docs/architecture.md:25]: Plus inspect the source code?

Comment (Jakic007) [reply docs/architecture.md:25]: @chriseth That's what repository is for, right? UI just provides a link to the source code in the repository

Comment (FabijanC) [reply docs/architecture.md:25]: You mean by the fact that a link to the source code in the repository is generated if the checking says that the contract is indeed verified? That can be added.

Comment (chriseth) [reply docs/architecture.md:25]: Ah right!

**Pull request merged**